dev/ProgressBar/ProgressBar.cpp

```diff
@@ -0,0 +1,154 @@
+    }
+
+    m_progressBarIndicator.set(GetTemplateChildT<winrt::Rectangle>(L"ProgressBarIndicator", controlProtected));
+    if (auto progressBarIndicator = m_progressBarIndicator.get())
+    {
+        m_progressBarIndicatorRevoker = progressBarIndicator.Loaded(winrt::auto_revoke, { this, &ProgressBar::OnLayoutRootLoaded });
+    }
+
+    UpdateStates();
+}
+
+void ProgressBar::OnLoaded(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    // TODO: things
+}
+
+void ProgressBar::OnLayoutRootLoaded(const winrt::IInspectable&, const winrt::RoutedEventArgs&)
+{
+    // TODO: things
+}
+
+void ProgressBar::OnRangeBaseValueChanged(const winrt::DependencyObject& sender, const winrt::DependencyProperty& args)
+{
+    // NOTE: This hits when the Value property changes, because we called RegisterPropertyChangedCallback.
+
+    SetProgressBarIndicatorWidth();
+}
+
+void ProgressBar::OnIsIndeterminatePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    // NOTE: This hits when IsIndeterminate changes because we set MUX_PROPERTY_CHANGED_CALLBACK to true in the idl.
+
+    UpdateStates();
+}
+
+void ProgressBar::OnShowPausedPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::OnShowErrorPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    UpdateStates();
+}
+
+void ProgressBar::UpdateStates()
+{
+    if (ShowError())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Error", true);
+    }
+    else if (ShowPaused() && IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Error", true); // Paused-Indeterminate state same visual treatment as Error state
+    }
+    else if (ShowPaused())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Paused", true);
+    }
+    else if (IsIndeterminate())
+    {
+        UpdateWidthBasedTemplateSettings();
+        winrt::VisualStateManager::GoToState(*this, L"Indeterminate", true);
+    }
+    else if (!IsIndeterminate())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"Determinate", true);
+    }
+}
+
+void ProgressBar::SetProgressBarIndicatorWidth()
+{
+    if (auto progressBarIndicator = m_progressBarIndicator.get())
+    {
+        double progressBarWidth = m_layoutRoot.get().ActualWidth();
+        double maximum = Maximum();
+        double minimum = Minimum();
+        double increment = 0;
+        
+        increment = progressBarWidth / (maximum - minimum);
+        progressBarIndicator.Width(increment * (Value() - minimum));
+    }
+}
+
+void ProgressBar::UpdateWidthBasedTemplateSettings()
+{
+    auto const templateSettings = winrt::get_self<::ProgressBarTemplateSettings>(TemplateSettings());
+    auto progressBar = m_layoutRoot.get();
+    auto progressBarIndicator = m_progressBarIndicator.get();
+
+    auto const [width, height] = [progressBar]()
```

Comment:
>progressBar [](start = 34, length = 11)

since this lambda is the only place that this variable is used you can move its declaration into the capture block.

[progressBar = m_layouRoot.get()]

and delete the above local.